src/miner.cpp

```diff
@@ -342,8 +362,12 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const MinerAddre
         // This vector will be sorted into a priority queue:
         vector<TxPriority> vecPriority;
         vecPriority.reserve(mempool.mapTx.size());
-        for (CTxMemPool::indexed_transaction_set::iterator mi = mempool.mapTx.begin();
-             mi != mempool.mapTx.end(); ++mi)
+
+        // If we're given a coinbase tx, it's been precomputed, so its fees are zero,
+        // so we can't include any mempool transactions; this will be an empty block.
+        if (next_coinbase_tx == nullptr) for (
```

Comment:
This looks really weird, but our (bitcoin core's) coding standards let us write `if (condition) statement` (all on one line). I did it this way because otherwise the indenting of this huge following block of code changes.

The reason for this `if` guard here is that if `next_coinbase_tx` is being passed in (non-null pointer), then we're forming an empty block, so we want to skip adding transactions to the block.